controllers/releaseplan/adapter.go

```diff
@@ -74,3 +76,27 @@ func (a *adapter) EnsureOwnerReferenceIsSet() (controller.OperationResult, error
 
 	return controller.ContinueProcessing()
 }
+
+// EnsurePairingIsSet is an operation that will ensure that the ReleasePlan has updated pairing
+// information in its status.
+func (a *adapter) EnsurePairingIsSet() (controller.OperationResult, error) {
+	a.logger.Info("in ReleasePlan adapter EnsurePairingIsSet")
+	// If an error occurs getting the ReleasePlanAdmission, mark the ReleasePlan as unmatched
+	releasePlanAdmission, _ := a.loader.GetReleasePlanAdmissionFromReleasePlan(a.ctx, a.client, a.releasePlan)
+
+	copiedReleasePlan := a.releasePlan.DeepCopy()
+	patch := client.MergeFrom(a.releasePlan.DeepCopy())
+
+	if releasePlanAdmission == nil {
+		a.releasePlan.MarkUnmatched()
+	} else {
+		a.releasePlan.MarkMatched(releasePlanAdmission)
+	}
+
+	if reflect.DeepEqual(copiedReleasePlan.Status.ReleasePlanAdmission, a.releasePlan.Status.ReleasePlanAdmission) {
```

Comment:
You probably don't need this as I doubt any patch is done if there's no diff.

Reply:
There would be a diff because the lastTransitionTime on the condition is changed. That is why I checked if the ReleasePlanAdmission has changed or not

Reply:
I don't see that as a problem but sure.

Reply:
If there is no change at all, I don't think the lastTransitionTime should update. That is why I put the check

Reply:
You are not using pointers, so wondering if you can just save the `status.RPA` in a variable without making a deepCopy and compare it later.